### PR TITLE
Week18 - MongoDB Ida Halling

### PR DESCRIPTION
**Pull request by id4h4lling**
https://week18-project-mongo.herokuapp.com/

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,35 +1,110 @@
...
+    res.send(singleSong)
+  } else {
+    const singleSong = await Song.find({ artistName: artistName })
+    res.send(singleSong)
+  }
+})
+
+// http://localhost:8080/songs/song?artistName=Marshmello&trackName=Happier  - en låt från en artist
+// http://localhost:8080/songs/song?trackName=Happier  - en låt med ett givet namn
+// http://localhost:8080/songs/song?artistName=Marshmello- alla låtar från en artist FUNKAR
```
Comment (Thereese) [inline server.js:105]: Maybe delete if it´s not used?

File: server.js [hunk 2]
```diff
@@ -1,35 +1,110 @@
...
+app.use(express.json())
 
 // Start defining your routes here
-app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
-});
+
+app.get('/', (req, res) => {
+  res.send(getEndpoints(app))
+})
```
Comment (Thereese) [inline server.js:58]: This part makes it so much easier to understand, nice!

File: server.js [hunk 3]
```diff
@@ -1,35 +1,110 @@
...
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
-mongoose.Promise = Promise;
+import express from 'express'
+import cors from 'cors'
+import mongoose from 'mongoose'
+import dotenv from 'dotenv'
+import getEndpoints from 'express-list-endpoints'
+
+import topMusicData from './data/top-music.json'
+import { reset } from 'nodemon'
```
Comment (Thereese) [inline server.js:8]: Had to google this, convenient!:)

File: server.js [hunk 4]
```diff
@@ -1,35 +1,110 @@
...
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
-
-const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/project-mongo";
-mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true });
-mongoose.Promise = Promise;
+import express from 'express'
+import cors from 'cors'
+import mongoose from 'mongoose'
+import dotenv from 'dotenv'
```
Comment (Thereese) [inline server.js:4]: Also new for me, I think! Don´t really understand after googling, might be tired, haha:) Maybe you can tell me more about  it?

**Review by Thereese — APPROVED** [review]
Everything is working and I can easily read your code. Great that you made comments about what the different endpoints do. Good job, Halling!🤩

File: server.js (hunk 4 above)
Comment (id4h4lling) [reply server.js:4]: It's for storing environment variables that can store secrets, Damien talks about it in the material. I just tried it out but it wasn't needed for this project in particular. The environment variables is stored in another folder in the project - so they can be reached from all over, the ones i tried stored secrets and are not pushed to github.

**Pull request closed without merging**